crowbar_framework/app/models/service_object.rb

```diff
@@ -1421,6 +1432,24 @@ def proposal_dependencies(role)
     []
   end
 
+  def expand_items_in_elements(elements)
+    # expand items in elements that are not nodes
+    expanded_new_elements = {}
+    failures = []
+    elements.each do |role_name, nodes|
+      expanded_new_elements[role_name], failures = expand_nodes_for_all(nodes)
```

Comment:
You're overriding `failures` here: if there's a failure for the first element, but not the second element, this will go unnoticed.

Reply:
Good point, I should break here like in the original code when we find an error 👍

Reply:
plus we have no message to return :/